components/datadog/apps/nginx/ecsFargateReadonly.go

```diff
@@ -0,0 +1,112 @@
+package nginx
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	fakeintakeComp "github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+	ecsComp "github.com/DataDog/test-infra-definitions/components/ecs"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	ecsClient "github.com/DataDog/test-infra-definitions/resources/aws/ecs"
+	classicECS "github.com/pulumi/pulumi-aws/sdk/v6/go/aws/ecs"
+	"github.com/pulumi/pulumi-awsx/sdk/v2/go/awsx/ecs"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func FargateReadonlyAppDefinition(e aws.Environment, clusterArn pulumi.StringInput, apiKeySSMParamName pulumi.StringInput, fakeIntake *fakeintakeComp.Fakeintake, opts ...pulumi.ResourceOption) (*ecsComp.Workload, error) {
+	namer := e.Namer.WithPrefix("nginx").WithPrefix("fg-readonly")
+
+	opts = append(opts, e.WithProviders(config.ProviderAWS, config.ProviderAWSX))
+
+	EcsFargateComponent := &ecsComp.Workload{}
+	if err := e.Ctx().RegisterComponentResource("dd:apps", namer.ResourceName("grp"), EcsFargateComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(EcsFargateComponent))
+
+	serverContainer := &ecs.TaskDefinitionContainerDefinitionArgs{
+		Name:  pulumi.String("nginx"),
+		Image: pulumi.String("ghcr.io/datadog/apps-nginx-server:main"),
```

Comment:
In order to get better test reproducibility, the use of floating `main` tag is now deprecated and should be replaced by “fixed” tags:

```suggestion
		Image: pulumi.String("ghcr.io/datadog/apps-nginx-server:" + apps.Version),
```